openvdb_ax/openvdb_ax/test/backend/TestFunctionTypes.cc

```diff
@@ -1699,29 +1662,43 @@ TestFunctionTypes::testIRFunctions()
         (const std::vector<llvm::Value*>& args,
          llvm::IRBuilder<>& _B) -> llvm::Value*
     {
+        // NOTE: GTest's ASSERT_* macros create a 'return;' statement
+        // which errors here. Exceptions are being used instead to
+        // exit early.
+
         // test the builder is pointing to the correct location
-        CPPUNIT_ASSERT(&B != &_B);
+        if (&B == &_B)
+        {
+            throw std::runtime_error("Builder should be different");
+        }
```

Comment:
Some opinionated changes: GTest `ASSERT_*` macros can't be used in a non-void function because they expand to a void return statement (I suppose Google wanted to support code bases with C++ exceptions disabled). I've replaced the assertions with equivalent if-statements and explicit C++ exceptions. I've confirmed that GTest reports a failure as expected when these are thrown here.